### PR TITLE
fix(pattern validation): Prevent wrong white spaces in business data

### PR DESCRIPTION
**Pull request by typecastcloud**
## Description

1. Remove **trim()** from validation because it prevents validating against leading/trailing white spaces in customer input.
2. Update patterns
   - EORI:
      - Worldwide:
         - Following: https://taxation-customs.ec.europa.eu/
         - Starting with ISO 3166 Country Prefix (e.g., “DE” for Germany)
         - Followed by 1-15 lower/upper case letters or numbers
         - No white spaces allowed
   - VAT_ID:
      - Worldwide: Increased length for Swiss VAT ID (example: CHE-116.281.710 MWST)
      - IN: Increased length to 15 for valid GST https://services.gst.gov.in/services/searchtp
   - VIES:
      - Added generic pattern for EUROPE
         - Following: https://ec.europa.eu/taxation_customs/vies/checkVatService.wsdl
   - Commercial Reg. Number:
      - Worldwide:
          - Lower limit reduced to 4 characters
          - Upper limit stays the same (up to 21 character)
          - Allow white spaces (not at beginning or end)
          - Allow '.' dot
          - Allow '/' slash
          - Allow '-' hyphen
   - LEI: no change
   - Other patterns:
      - General change: Prevent leading and trailing white space if allowed otherwise
4. Restructure test setup
5. Added tests

@oyo
Alternativ for 1. removal of trim before validation: trim on data saving/sending

## Why

Currently validation for leading and trailing white spaces is not applied (trimming happens before validation). Some patterns do not prevent this type of input. Faulty data causes issues in connected services.

## Issue

Refs: #190 #191 #192

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [ ] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/types/Patterns.ts [hunk 1]
```diff
@@ -23,8 +23,55 @@ export const Patterns = {
   CITY: /^[A-ZÀ-ÿ0-9Śął](([ .'-]|\. )?[A-Za-zÀ-ÿ0-9Śął]{1,40}){1,10}$/,
   STREET:
```
Comment (jjeroch) [inline src/types/Patterns.ts:24]: Can we delete the "Street" pattern? It seems to be not used at all

**Review by jjeroch — CHANGES_REQUESTED** [review]

**Comment by typecastcloud** [comment]
I still need to add test cases.

**Comment by typecastcloud** [comment]
COMMERCIAL_REG_NUMBER **pattern** seems to be wrong. In germany a valid example would be 'HRA 12345'
INDIAN vat id **pattern** seems to be wrong. India uses GSTIN (Goods and Services Tax Identification Number) with 15 characters, right?
Worldwide EORI pattern is also wrong. No spaces and "up to" 17 characters is what I found.
- https://www.dpd.com/de/en/faq/eori-nummer/
- https://design.tax.service.gov.uk/hmrc-design-patterns/eori-numbers/

File: src/types/Patterns.ts (hunk 1 above)
Comment (oyo) [reply src/types/Patterns.ts:24]: @jjeroch it's used for a mandatory field in the company data page

Comment (typecastcloud) [reply src/types/Patterns.ts:24]: @oyo @jjeroch Seems like streetHouseNumberPattern is not used.

**Comment by typecastcloud** [comment]
@jjeroch @oyo ready for review now.

Additional concerns I have:
1. German CRN = "Handelsregister"? Something else? I do not understand where the pattern is coming from.
2. Should I also create tests for other patterns? Some of them are very limited for no reason. Like city name having a maximum of 10 characters.

File: src/types/Patterns.test.ts [hunk 2]
```diff
@@ -110,14 +37,56 @@ const validate = (data, check) => {
...
+  validatePattern('DE', 'VIES', VIES_TEST_DATA)
+  validatePattern('FR', 'VIES', VIES_TEST_DATA)
+  validatePattern('Worldwide', 'VIES', VIES_TEST_DATA)
+
+  // Commercial Registration Number
+  validatePattern('DE', 'COMMERCIAL_REG_NUMBER', CRN_TEST_DATA)
+  validatePattern('FR', 'COMMERCIAL_REG_NUMBER', CRN_TEST_DATA)
+  validatePattern('MX', 'COMMERCIAL_REG_NUMBER', CRN_TEST_DATA)
+  validatePattern('IN', 'COMMERCIAL_REG_NUMBER', CRN_TEST_DATA)
+  validatePattern('Worldwide', 'COMMERCIAL_REG_NUMBER', CRN_TEST_DATA)
```
Comment (oyo) [inline src/types/Patterns.test.ts:90]: this block only tests the validatePattern function and should be wrapped in a test case function.
```
it('validates patterns', () => {
  validatePattern(...)
  ...
})
```

**Review by oyo — CHANGES_REQUESTED** [review]
wrap collection of single tests in test case function for better structure

File: src/types/Patterns.test.ts (hunk 2 above)
Comment (typecastcloud) [reply src/types/Patterns.test.ts]: Test case is created inside the validate*Pattern function.

I refactored the code with describe grouping and made it more expendable for more countries.

**Comment by typecastcloud** [comment]
I guess I have to split up the PR into smaller changes. I'll create and reopen.

**Comment by typecastcloud** [comment]
Will be split up in smaller PRs for better reviewability.

File: src/types/Patterns.ts (hunk 1 above)
Comment (oyo) [reply src/types/Patterns.ts:51]: @typecastcloud yes - please remove

**Review by oyo — APPROVED** [review]
lgtm now :)

**Pull request merged**